### PR TITLE
std.algo.cmp.cmp doco

### PR DESCRIPTION
**Pull request by burner**
Params, Returns
Also removed some incomplete sentences

### CODE REVIEW

File: std/algorithm/comparison.d [hunk 1]
```diff
@@ -546,6 +544,17 @@ elements.
 
 If the ranges are strings, $(D cmp) performs UTF decoding
 appropriately and compares the ranges one code point at a time.
+
+Params:
+    pred = The predicated used for comparision.
```
Comment (quickfur) [inline std/algorithm/comparison.d]: Typo: "comparison", not "comparision".
Typo: "predicate", not "predicated".

File: std/algorithm/comparison.d [hunk 2]
```diff
@@ -546,6 +544,17 @@ elements.
...
+    r1 = The first range.
+    r2 = The second range.
+
+Returns:
+    0 if both all ranges are equal. -1 if $(D r1) is less than $(D r2)
```
Comment (quickfur) [inline std/algorithm/comparison.d]: Grammar: "if both ranges are *entirely* equal".

File: std/algorithm/comparison.d [hunk 3]
```diff
@@ -546,6 +544,17 @@ elements.
...
 If the ranges are strings, $(D cmp) performs UTF decoding
 appropriately and compares the ranges one code point at a time.
+
+Params:
+    pred = The predicate used for comparison.
+    r1 = The first range.
+    r2 = The second range.
+
+Returns:
+    0 if both all ranges are entirely equal. -1 if $(D r1) is less than $(D r2)
```
Comment (quickfur) [inline std/algorithm/comparison.d]: Grammar: "both ranges", not "both all ranges".

Comment (quickfur) [inline std/algorithm/comparison.d]: Also, "r1 is less than r2" may not be very obvious when r1 and r2 are ranges. What about "if the first differing element of r1 is less than the corresponding element of r2"? Ditto for the greater than case.

Comment (yazd) [inline std/algorithm/comparison.d]: I think it's better to drop 'entirely' from 'if both ranges are entirely equal.'

**Comment by quickfur** [comment]
LGTM, thanks!

**Comment by quickfur** [comment]
Auto-merge toggled on

**Comment by burner** [comment]
thanks

File: std/algorithm/comparison.d [hunk 4]
```diff
@@ -546,6 +544,18 @@ elements.
...
 If the ranges are strings, $(D cmp) performs UTF decoding
 appropriately and compares the ranges one code point at a time.
+
+Params:
+    pred = The predicate used for comparison.
+    r1 = The first range.
+    r2 = The second range.
+
+Returns:
+    0 if both ranges compare equal. -1 if the first differing element of $(D
```
Comment (JakobOvrum) [inline std/algorithm/comparison.d]: compare equal *according to `pred`*. IMO, it looks a little confusing when the following two sentences are explicit about it.

**Comment by burner** [comment]
thanks

**Pull request merged**